meson.build

```diff
@@ -1,47 +1,129 @@
-project('openinput-cm3', 'c')
+project('openinput-cm3', 'c', license: 'MIT')
 
-sources = [
-        'src/main.c'
+if not meson.is_cross_build()
+        error('Cross-compiling is required. Please run meson with the \'--cross-file targets/family/my-target-family.txt\' \'--cross-file targets/my-target.txt\' flag')
+endif
+
+# ---------------------------------------
+# Target independent sources & flags
+# ---------------------------------------
+
+source_dirs = [
+	'src',
+	'external/tinyusb/src',
+	'external/tinyusb/src/class/hid',
+	'external/tinyusb/src/common',
+	'external/tinyusb/src/device'
+]
+
+include_dirs = [
+	'src/target',
+	'external/tinyusb/src/osal'
 ]
 
 c_flags = [
-        '-Os'
+	'-Os',
+	'-nostdlib',
+	'-nostartfiles',
+	'-ffunction-sections',
+	'-fdata-sections',
+	'-std=gnu99',
+	'-Wpointer-arith',
+	'-Wundef',
+	'-Werror'
 ]
 
 ld_flags = [
-        '--specs=nosys.specs',
-        '-nostartfiles',
-        '-Wl,--gc-sections',
-        '-T' + meson.get_cross_property('linker') + '.ld'
+	'-lm',
+	'-lc',
+	'-lgcc',
+	'-lnosys',
+	'-nostdlib',
+	'-nostartfiles',
+	'-fdata-sections',
+	'-ffunction-sections',
+	'--specs=nano.specs',
+	'--specs=nosys.specs',
+	'-Wl,--gc-sections',
+	'-T../build-targets/ld/' + meson.get_cross_property('linker') + '.ld'
 ]
 
-if not meson.is_cross_build()
-        error('Cross-compiling is required. Please run meson with the \'--cross-file targets/family/my-target-family.txt\' \'--cross-file targets/my-target.txt\' flag')
+# ---------------------------------------
+# Target dependent sources & flags
+# ---------------------------------------
+
+target_family = meson.get_cross_property('family')
+
+if target_family == 'stm32f1'
+	include_dirs += [
+		'external/cmsis-5/CMSIS/Core/Include',
+		'external/cmsis-device-stm32f1/Include'
+	]
+	source_dirs += [
+		'src/drv/mcu/stm32f1',
+		'external/tinyusb/src/portable/st/stm32_fsdev'
+	]
+else
+	error('This family is not supported.')
+endif
+
+target = meson.get_cross_property('target')
+
+if target == 'blue-pill'
+	c_flags += [
+		'-DSTM32F103xB',
+		'-DHSE_VALUE=8000000UL'
+	]
```

Comment:
I don't love these flags being added here, but I'm not sure how else to do it for now.

Reply:
Yes, put it in the cross files.

Reply:
when i tried to add c_args to both cross-files, they overwrite eachother, do i put them in another array? or did i overlook it and can actualy do it?